### PR TITLE
[Draft]Testing Hapigen Output for NI-Tclk

### PR DESCRIPTION
**Pull request by IPetersNI**
### What does this Pull Request accomplish?

This Pull Request test the output of the hapigen files for NI-Tclk. This is paired with the Pull Request in azure located here: https://ni.visualstudio.com/DevCentral/_git/ni-central/pullrequest/358280.

### Why should this Pull Request be merged?

This Pull Request should not be merge as it is a draft request. This is only to use for building and seeing the differences within the generated files/built files.

### What testing has been done?
Testing has been done manually using local builds and updates for metadata and generated files.

### CODE REVIEW

File: generated/nitclk/nitclk.proto [hunk 1]
```diff
@@ -9,24 +9,17 @@ syntax = "proto3";
...
 
-package nitclk_grpc;
+package niTClk_grpc;
 
 import "session.proto";
 
 service NiTClk {
   rpc ConfigureForHomogeneousTriggers(ConfigureForHomogeneousTriggersRequest) returns (ConfigureForHomogeneousTriggersResponse);
   rpc FinishSyncPulseSenderSynchronize(FinishSyncPulseSenderSynchronizeRequest) returns (FinishSyncPulseSenderSynchronizeResponse);
-  rpc GetAttributeViReal64(GetAttributeViReal64Request) returns (GetAttributeViReal64Response);
```
Comment (reckenro) [inline generated/nitclk/nitclk.proto:21]: Guessing all these Get and SetAttribute methods that are missing are marked 'codgen_method': 'no' right now in hapigen. Need to remove those entries so these functions generate again.

File: generated/nitclk/nitclk.proto [hunk 2]
```diff
@@ -9,24 +9,17 @@ syntax = "proto3";
 option java_multiple_files = true;
 option java_package = "com.ni.grpc.tclk";
 option java_outer_classname = "NiTClk";
-option csharp_namespace = "NationalInstruments.Grpc.TClk";
```
Comment (reckenro) [inline generated/nitclk/nitclk.proto:12]: We can add this to hapigen's grpc_device/global_config/addition.hapigen like 'chsarp_namespace': NationalInstruments.Grpc.TClk' to get this generating like it used to.

File: generated/nitclk/nitclk.proto [hunk 3]
```diff
@@ -9,24 +9,17 @@ syntax = "proto3";
...
+option csharp_namespace = "NationalInstruments.Grpc.Tclk";
 
-package nitclk_grpc;
```
Comment (reckenro) [inline generated/nitclk/nitclk.proto:14]: I'm not sure why this changed. This should just be based on config.py's `namespace_component` which hasn't changed. Maybe after fixing the chsarp_namespace and rebuilding it'll clear this up.

File: generated/nitclk/nitclk.proto [hunk 4]
```diff
@@ -39,7 +32,6 @@ enum NiTClkAttribute {
   NITCLK_ATTRIBUTE_EXPORTED_SYNC_PULSE_OUTPUT_TERMINAL = 2;
   NITCLK_ATTRIBUTE_START_TRIGGER_MASTER_SESSION = 3;
   NITCLK_ATTRIBUTE_REF_TRIGGER_MASTER_SESSION = 4;
-  NITCLK_ATTRIBUTE_SCRIPT_TRIGGER_MASTER_SESSION = 5;
```
Comment (reckenro) [inline generated/nitclk/nitclk.proto:42]: Missing. I think in hapigen right now it's marked with 'codegen_method': 'no' in grpc_device/attributes/attributes.hapigen. You can clear that entry and this should re-generate properly.

File: source/codegen/metadata/nitclk/functions.py [hunk 5]
```diff
@@ -1,606 +1,15 @@
...
-                'direction': 'out',
-                'documentation': {
-                    'description': 'The value that you are getting'
-                },
-                'grpc_type': 'string',
-                'name': 'value',
-                'size': {
-                    'mechanism': 'ivi-dance',
-                    'tags': [
-                        'strlen-bug'
```
Comment (reckenro) [inline source/codegen/metadata/nitclk/functions.py:421]: For 'GetAttributeVIString', we'll need this 'strlen-bug' tag back. To add it, in hapigen's grpc_device/functions/functions.hapigen update 'get attribute vi string' to have the following:

```
'parameters': {
  'value': {
     'array': {
                    'tags': ['strlen-bug'],
     },
  }

File: source/codegen/metadata/nitclk/nitclk.proto [hunk 6]
```diff
@@ -0,0 +1,195 @@
+
```
Comment (reckenro) [inline source/codegen/metadata/nitclk/nitclk.proto:1]: Same as the comment in nidigital, we're not including the proto files generated in hapigen here yet. You can just delete this file out of source/codegen/metadata/nitclk

File: generated/nitclk/nitclk_library.cpp [hunk 7]
```diff
@@ -6,7 +6,7 @@
 #include "nitclk_library.h"
 
 #if defined(_MSC_VER)
-static const char* kLibraryName = "nitclk_64.dll";
+static const char* kLibraryName = "niTClk_64.dll";
```
Comment (reckenro) [inline generated/nitclk/nitclk_library.cpp:9]: Note: I think this case change for the dll is OK and correct based on the base hapigen metadata.

**Comment by dmondrik** [comment]
These changes should be rendered obsolete by my PR #754
This PR should be abandoned.

**Pull request closed without merging**